packages/playwright-core/src/server/bidi/bidiBrowser.ts

```diff
@@ -458,8 +459,10 @@ function getProxyConfiguration(proxySettings?: types.ProxySettings): bidi.Sessio
     default:
       throw new Error('Invalid proxy server protocol: ' + proxySettings.server);
   }
-  if (proxySettings.bypass)
-    proxy.noProxy = proxySettings.bypass.split(',');
+  const bypass = proxySettings.bypass ?? process.env.PLAYWRIGHT_PROXY_BYPASS_FOR_TESTING;
+  if (bypass)
+    proxy.noProxy = bypass.split(',');
+  proxy.noProxy = ['<-loopback>', ...proxySettings.bypass?.split(',') ?? []];
```

Comment:
Looks like it adds it regardless of the PLAYWRIGHT_PROXY_BYPASS_FOR_TESTING variable